verification/utils/slices/slices_contents.gobra

```diff
@@ -0,0 +1,194 @@
+	fold  AbsSliceC_Bytes(s[start:end], 0, len(s[start:end]), contents)
+}
+
+
+// The slice axiomatization is very incomplete.
+// The following body cannot be verified yet.
+ghost
+requires 0 <= p
+requires 0 <= start && start <= end && end <= cap(s)
+requires len(s[start:end]) <= cap(s)
+requires acc(AbsSliceC_Bytes(s[start:end], 0, len(s[start:end]), contents), p)
+ensures  acc(AbsSliceC_Bytes(s, start, end, contents), p)
+decreases
+func UnsliceC_Bytes(s []byte, start int, end int, contents seq[byte], p perm) /*{
+	unfold AbsSliceC_Bytes(s[start:end], 0, len(s[start:end]), contents)
+	assert 0 <= start && start < end && end <= cap(s)
+	assert len(contents) == end - start
+	assert forall i int :: 0 <= i && i < len(s[start:end]) ==> &s[start:end][i] == &s[start + i]
+	assert forall i int :: 0 <= i && i < len(s[start:end]) ==> &s[start:end][i] == &s[start + i]
+	assert forall i int :: 0 <= i && i < len(s[start:end]) ==> acc(&s[start:end][i])
+	// failling assertion:
+	assert forall i int :: start <= i && i < end ==> acc(&s[i])
+	assert forall i int :: start <= i && i < end ==> s[i] == contents[i - start]
+	fold   AbsSliceC_Bytes(s, start, end, contents)
+}*/
+
+
+/*
+ghost
+requires 0 < p
+requires 0 <= start && start <= end && end <= cap(s)
+requires len(s[start:end]) <= cap(s)
+requires acc(AbsSlice_Bytes(s[start:end], 0, len(s[start:end])), p)
+ensures  acc(AbsSlice_Bytes(s, start, end), p)
+ensures  old(unfolding acc(AbsSlice_Bytes(s[start:end], 0, len(s[start:end])), p)
+decreases
+func UnsliceC_Bytes(s []byte, start int, end int, p perm) {
+	unfold acc(AbsSlice_Bytes(s[start:end], 0, len(s[start:end])), p)
+	assert 0 <= start && start <= end && end <= cap(s)
+	assert forall i int :: 0 <= i && i < len(s[start:end]) ==> acc(&s[start:end][i], p)
+	assert forall i int :: 0 <= i && i < len(s[start:end]) ==> &s[start:end][i] == &s[start + i]
+	assert forall i int :: 0 <= i && i < len(s[start:end]) ==> &s[start:end][i] == &s[start + i]
+
+	invariant 0 <= j && j <= len(s[start:end])
+	invariant forall i int :: j <= i && i < len(s[start:end]) ==> acc(&s[start:end][i], p)
+	invariant forall i int :: 0 <= i && i < len(s[start:end]) ==> &s[start:end][i] == &s[start + i]
+	invariant forall i int :: 0 <= i && i < len(s[start:end]) ==> &s[start:end][i] == &s[start + i]
+	invariant forall i int :: start <= i && i < start+j ==> acc(&s[i], p)
+	decreases len(s[start:end]) - j
+	for j := 0; j < len(s[start:end]); j++ {
+		assert forall i int :: start <= i && i < start+j ==> acc(&s[i], p)
+		assert &s[start:end][j] == &s[start + j]
+		assert acc(&s[start + j], p)
+		assert forall i int :: start <= i && i <= start+j ==> acc(&s[i], p)
+	}
+	fold acc(AbsSlice_Bytes(s, start, end), p)
+  }*/
+
+/** Auxiliary definitions of Bytes **/
+
+ghost
+requires size >= 0
+ensures len(res) == size
+ensures forall i int :: 0 <= i && i < size ==> res[i] == byte(0)
+decreases
+pure func NewSeq_Bytes(size int) (res seq[byte])
+
+/** End of auxiliary definitions of Bytes **/
```

Comment:
I would drop this completely or move it to the method `NewSeqBytes` in this package (https://github.com/viperproject/VerifiedSCION/blob/master/verification/utils/seqs/seqs.gobra)